src/hotspot/share/prims/jvmtiEnvThreadState.cpp

```diff
@@ -258,19 +258,16 @@ class GetCurrentLocationClosure : public HandshakeClosure {
     JavaThread *jt = target->as_Java_thread();
     ResourceMark rmark; // jt != Thread::current()
     RegisterMap rm(jt, false);
-    // There can be a race condition between a VM_Operation reaching a safepoint
+    // There can be a race condition between a handshake
     // and the target thread exiting from Java execution.
     // We must recheck the last Java frame still exists.
     if (!jt->is_exiting() && jt->has_last_Java_frame()) {
       javaVFrame* vf = jt->last_java_vframe(&rm);
-      assert(vf != NULL, "must have last java frame");
```

Comment:
The code before we converted to handshakes also had this assert.

The pre-handshake code did the work in the doit() function for the
VM_GetCurrentLocation VM-op. This makes me wonder if we always
had frames here when this was previously done via VM-op? And that
makes me wonder whether handshakes is doing something different
so we don't always have a frame here?

Reply:
The differences is 8253180 (JDK 16), which turns return polls into branches to SafepointBlob instead of going via signal handler.
When setting up the last java frame in the SafepointBlob we get different result than before.
To look at that potential bug I opened 8263576.